### DISCUSSION TITLE
TypeError: undefined is not an object when filtering

### DISCUSSION THREAD

**Discussion by JanaS87** [neuefische/questions#713] [Web] [answered]
Hello everyone,

I am currently trying to filter the food based on health information so that only the "good" food is listed.

For this, I have written a function that searches the information using .some(). This had to be slightly adapted for the diseases.

When I then test the program and call "Walter", I get this error:

![Bildschirmfoto 2023-08-31 um 14 31 32](https://github.com/neuefische/questions/assets/136725332/f7acf40b-d16e-450d-acfd-709d69fc0366)

When I create a new cat and, for example, tick CNI for diseases, create the cat, and then try to access its profile, I get the same error, so it must have something to do with the way I handle the diseases.

This is my PR: [PR](https://github.com/JanaS87/capstone-project/pull/25)

Do you have any ideas or suggestions on how I can solve this? 
Anything that helps is warmly welcome.

**Comment by mbosselmann** [comment, marked as answer]
Hi Jana,
I tried this out and everything works like it should. Have you deleted the local storage before trying this out? Maybe there is old data stored in there. 👀

**Reply by JanaS87** [reply]
After deleting the cache in firefox it's running, too.
